edge-runtime/shim/node.js

```diff
@@ -1,16 +1,21 @@
 // NOTE: This is a fragment of a JavaScript program that will be inlined with
 // a Webpack bundle. You should not import this file from anywhere in the
 // application.
-import { AsyncLocalStorage } from 'node:async_hooks'
-
 import { createRequire } from 'node:module' // used in dynamically generated part
-import process from 'node:process'
 
 import { registerCJSModules } from '../edge-runtime/lib/cjs.ts' // used in dynamically generated part
 
-globalThis.process = process
+if (typeof process === 'undefined') {
+  globalThis.process = (await import('node:process')).default
+}
+
+if (typeof AsyncLocalStorage === 'undefined') {
+  globalThis.AsyncLocalStorage = (await import('node:async_hooks')).AsyncLocalStorage
+}
 
-globalThis.AsyncLocalStorage = AsyncLocalStorage
+if (typeof Buffer === 'undefined') {
+  globalThis.Buffer = (await import('node:buffer')).Buffer
+}
```

Comment:
this is fun one - Buffer is not globally defined in deno 1.39.0, 1.46.3 or 2.2.4 (from tested versions), but is available in 2.4.2